app/controllers/bulk_ingest_controller.rb

```diff
@@ -90,8 +60,124 @@ def source_metadata_id_from_path
       base_path if valid_remote_identifier?(base_path)
     end
 
+    def browse_everything_params
+      return {} unless params.key?("browse_everything")
+      @browse_everything_params ||= params["browse_everything"]
+    end
+
+    def browse_everything_uploads
+      return [] unless browse_everything_params.key?("uploads")
+      @browse_everything_uploads ||= browse_everything_params["uploads"]
+    end
+
+    def find_upload(upload_id)
+      uploads = BrowseEverything::Upload.find_by(uuid: upload_id)
+      uploads.first
+    end
+
+    def new_uploads
+      @new_uploads ||= begin
+                         browse_everything_uploads.map do |upload_id|
+                           find_upload(upload_id)
+                         end
+                       end
+    end
+
+    def ingest_multi_volume_works(change_set_persister)
+      new_uploads.each do |upload|
+        file_tree = {}
```

Comment:
You can make this `Hash.new([])` so you don't have to do all that || business below, just `file_tree[upload_file.container_id] += [new_pending_upload]`, I think.